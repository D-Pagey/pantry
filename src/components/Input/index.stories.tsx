import React from 'react';
import { Story } from '@storybook/react/types-6-0';
import { Input, InputProps } from '.';

export default { title: 'Input', component: Input };

const Template: Story<InputProps> = (args) => <Input {...args} />;

export const Primary = Template.bind({});

Primary.args = {
    disabled: false,
    name: 'test',
    onBlur: () => null,
    onChange: () => null,
    testId: 'storybookInput',
    placeholder: 'test',
    value: ''
};
