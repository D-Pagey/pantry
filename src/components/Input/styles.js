import styled from 'styled-components/macro';
import { colours } from '../../tokens';

export const Input = styled.input`
    border: 0;
    border-bottom: 2px solid ${colours.darkGreen100};
    border-radius: 2px;
    box-sizing: border-box;
    font-size: 1rem;
    margin: ${({ margin }) => margin};
    padding: 12px;
    width: 100%;
`;
