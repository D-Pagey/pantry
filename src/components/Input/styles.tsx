import styled from 'styled-components/macro';
import { colours } from '../../tokens';

export const Wrapper = styled.div`
    display: flex;
    flex-direction: column;
    padding: 0 0 1rem;
`;

export const Input = styled.input`
    border: 0;
    border-bottom: 2px solid ${colours.darkGreen100};
    border-radius: 2px;
    padding: 12px;
`;
