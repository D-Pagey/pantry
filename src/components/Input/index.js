import React from 'react';
import { func, string } from 'prop-types';
import * as S from './styles';

const Input = ({ error, label, name, onBlur, onChange, placeholder, testId, value }) => {
    return (
        <S.Wrapper>
            {label && <S.Label htmlFor={testId}>{label}</S.Label>}

            <S.Input
                id={testId}
                data-testid={testId}
                type="text"
                name={name}
                onBlur={onBlur}
                onChange={onChange}
                placeholder={placeholder}
                value={value}
            />

            {error && <S.Error>{error}</S.Error>}
        </S.Wrapper>
    );
};

Input.propTypes = {
    error: string,
    label: string,
    name: string,
    onBlur: func,
    onChange: func.isRequired,
    placeholder: string,
    testId: string,
    value: string.isRequired
};

Input.defaultProps = {
    error: '',
    label: '',
    name: '',
    onBlur: () => {},
    placeholder: '',
    testId: ''
};

export default Input;
