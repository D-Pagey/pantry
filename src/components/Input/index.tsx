import React from 'react';
import { func, string } from 'prop-types';
import * as S from './styles';

type props = {
    error: string;
    label: string;
    name: string;
    onBlur: Function;
    onChange: Function;
    placeholder: string;
    testId: string;
    value: string;
};

const Input = ({
    error,
    label,
    name,
    onBlur,
    onChange,
    placeholder,
    testId,
    value
}: props): JSX.Element => {
    return (
        <S.Wrapper>
            {label && <S.Label htmlFor={testId}>{label}</S.Label>}

            <S.Input
                id={testId}
                data-testid={testId}
                name={name}
                onBlur={onBlur}
                onChange={onChange}
                placeholder={placeholder}
                type="text"
                value={value}
            />

            {error && <S.Error>{error}</S.Error>}
        </S.Wrapper>
    );
};

Input.propTypes = {
    error: string,
    label: string,
    name: string,
    onBlur: func,
    onChange: func.isRequired,
    placeholder: string,
    testId: string,
    value: string.isRequired
};

Input.defaultProps = {
    error: '',
    label: '',
    name: '',
    onBlur: (): void => {},
    placeholder: '',
    testId: ''
};

export default Input;
