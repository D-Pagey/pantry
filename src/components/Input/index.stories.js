import React from 'react';
import { storiesOf } from '@storybook/react';
import Input from '.';

const props = {
    name: 'test',
    onBlur: () => {},
    onChange: () => {},
    testId: 'storybookInput',
    placeholder: 'test',
    value: ''
};

storiesOf('Input', module)
    .add('with label', () => <Input {...props} label="Name" />)
    .add('without label', () => <Input {...props} />)
    .add('with error', () => <Input {...props} error="Required" />);
