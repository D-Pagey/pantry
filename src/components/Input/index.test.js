import React from 'react';
import userEvent from '@testing-library/user-event';
import Input from '.';

const props = {
    name: 'test',
    onBlur: () => {},
    onChange: () => {},
    testId: 'testInput',
    value: ''
};

describe('Input component', () => {
    it('should render', () => {
        const { container } = render(<Input {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should show error', () => {
        const error = 'Required';
        const { getByText } = render(<Input {...props} error={error} />);
        const errorMessage = getByText(error);

        expect(errorMessage).toHaveStyleRule('color', 'red');
    });

    it('should show label', () => {
        const label = 'Name';
        const { getByText } = render(<Input {...props} label={label} />);
        getByText(label);
    });

    it('should handle change', () => {
        const onChange = jest.fn();
        const { getByTestId } = render(<Input {...props} onChange={onChange} />);

        userEvent.type(getByTestId('testInput'), 'testing onchange');

        expect(onChange).toHaveBeenCalled();
    });
});
