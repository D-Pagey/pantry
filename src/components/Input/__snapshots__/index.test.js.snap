// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render 1`] = `
.c0 {
  border: 0;
  border-bottom: 2px solid #348841;
  border-radius: 2px;
  box-sizing: border-box;
  font-size: 1rem;
  padding: 12px;
  width: 100%;
}

<input
  class="c0"
  data-testid="testInput"
  id="testInput"
  name="test"
  type="text"
  value=""
/>
`;
